src/python_bindings/py_data_stats.cpp

```diff
@@ -0,0 +1,166 @@
+#include "py_data_stats.h"
+
+#include "statistics/data_stats.h"
+
+namespace python_bindings {
+
+namespace py = pybind11;
+namespace mo = model;
+
+size_t PyDataStats::GetNumberOfValues(size_t index) const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).NumberOfValues(index);
+}
+
+size_t PyDataStats::GetNumberOfColumns() const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).GetNumberOfColumns();
+}
+
+std::vector<size_t> PyDataStats::GetNullColumns() const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).GetNullColumns();
+}
+
+std::vector<size_t> PyDataStats::GetColumnsWithNull() const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).GetColumnsWithNull();
+}
+
+std::vector<size_t> PyDataStats::GetColumnsWithUniqueValues() const {
+    return dynamic_cast<algos::DataStats&>(*algorithm_).GetColumnsWithUniqueValues();
+}
+
+size_t PyDataStats::Distinct(size_t index) const {
+    return dynamic_cast<algos::DataStats&>(*algorithm_).Distinct(index);
+}
+
+bool PyDataStats::IsCategorical(size_t index, size_t quantity) {
+    return dynamic_cast<algos::DataStats&>(*algorithm_).IsCategorical(index, quantity);
+}
+
+std::vector<std::vector<std::string>> PyDataStats::ShowSample(size_t start_row, size_t end_row,
+                                                              size_t start_col, size_t end_col,
+                                                              size_t str_len, size_t unsigned_len,
+                                                              size_t double_len) const {
+    const algos::DataStats& stat = GetAlgorithm<algos::DataStats>(algorithm_);
+    return stat.ShowSample(start_row, end_row, start_col, end_col, str_len, unsigned_len,
+                           double_len);
+}
+
+py::object GetValue(algos::Statistic stat) {
```

Comment:
```suggestion
py::object GetValue(algos::Statistic const& stat) {
```
Copying `Statistic` is a rather heavy operation